### ISSUE TITLE
 BertForMaskedLM Loss function 

### ISSUE THREAD

**Issue by lonelydancer** [huggingface/transformers#7136]
# ❓ Questions & Help

https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_bert.py

1)In BertForMaskedLM, why the loss function don't ignore the not masked tokens?
loss_fct = CrossEntropyLoss()  # -100 index = padding token
masked_lm_loss = loss_fct(prediction_scores.view(-1, self.config.vocab_size), labels.view(-1))
2) do i have to set the not masked token to -100 ?

## Details

**A link to original question on the forum/Stack Overflow**:

**Comment by lonelydancer** [comment]
torch.nn.CrossEntropyLoss(weight: Optional[torch.Tensor] = None, size_average=None, ignore_index: int = -100, reduce=None, reduction: str = 'mean')
